examples/project-delete/main.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	projectClient "github.com/openshift/client-go/project/clientset/versioned"
+	"github.com/pkg/errors"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func getOpenShiftProjectClient(kubeconfig string) (*projectClient.Clientset, error) {
+	// kubeconfig passed as parameter is coming from cmd line
+
+	// if kubeconfig not provided on cmd line then read it from the env
+	if kubeconfig == "" {
```

Comment:
@soltysh @juanvallejo this is going to happen for every CLI integration.  Here you have a very simple one.  I'd suggest using this to find the simplest way you can help them get a kubeconfig and all the "normal" CLI flags.

Reply:
@soltysh @deads2k There are utils that currently exist under kubectl/pkg/pluginutils that parse a kubeconfig file for plugins and attempt to handle global options: https://github.com/kubernetes/kubectl/blob/master/pkg/pluginutils/plugin_client.go#L36

I'd be in favor of updating these utils to be generic enough to be used outside of plugins.

Reply:
> I'd be in favor of updating these utils to be generic enough to be used outside of plugins.

This would also be a good place to pursue the FlagsStruct pattern that we are currently implementing with upstream printers. We could bind necessary flags to a struct, then be able to call methods on that struct to obtain a clientconfig or a restclient.

Reply:
First let's have them exist in kubectl but in k8s repo, then we'll extract them to the common repository we've discussed yesterday and we'll publish for external consumption. I don't want people rely on something that's unclear to me how is maintained.

Reply:
Currently we have all that login being part of the factory, looks like a perfect item to be extracted out, along the other flags.